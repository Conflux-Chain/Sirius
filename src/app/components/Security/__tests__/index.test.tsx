import React from 'react';
import { render } from '@testing-library/react';
import { Security } from '..';

describe('<Security  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<Security type="lv0"></Security>);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
