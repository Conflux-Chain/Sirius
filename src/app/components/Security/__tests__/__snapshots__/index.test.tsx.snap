// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Security  /> should match snapshot 1`] = `
.c0 .text {
  vertical-align: middle;
  font-size: 1rem;
  margin-left: 0.4286rem;
  line-height: 1.5714rem;
}

.c0 .img {
  background-color: transparent;
  height: 0.7143rem;
}

.c0 .img circle:last-child {
  -webkit-animation: blinker 1s linear infinite;
  animation: blinker 1s linear infinite;
}

.c0.high .img {
  background: linear-gradient(270deg,#7cd77b 0%,#e5f456 100%);
}

.c0.high .img .last-circle {
  -webkit-animation: none;
  animation: none;
}

.c0.high .text {
  color: #59bf9c;
}

.c0.medium .img {
  background: linear-gradient(270deg,#e5f456 0%,#ffd041 100%);
}

.c0.medium .text {
  color: #bfde5b;
}

.c0.low .img {
  background: linear-gradient(270deg,#ffd041 0%,#ec6057 100%);
}

.c0.low .text {
  color: #ffc324;
}

.c0.veryLow .img {
  background: linear-gradient(270deg,#ec6057 0%,#8d3a4a 100%);
}

.c0.veryLow .text {
  color: #ec6057;
}

.c0.not-available .img {
  display: none;
}

<span
  class="c0 sirius-security high"
>
  <svg
    class="img"
    viewBox="0 0 50 10"
  >
    <defs>
      <mask
        id="mask"
        x="0"
        y="0"
      >
        <rect
          fill="#fff"
          height="15"
          width="55"
          x="-5"
          y="-5"
        />
        <circle
          cx="5"
          cy="5"
          r="5"
        />
        <circle
          cx="18"
          cy="5"
          r="5"
        />
        <circle
          cx="31"
          cy="5"
          r="5"
        />
        <circle
          cx="43"
          cy="5"
          r="5"
        />
      </mask>
    </defs>
    <rect
      fill="#fff"
      height="10"
      mask="url(#mask)"
      width="50"
      x="0"
      y="0"
    />
  </svg>
  <span
    class="text"
  >
    High
  </span>
</span>
`;
